syntax = "proto3";

package ordermanagement;

// The Order Management Service
service OrderManagement {
  rpc CreateOrder (OrderRequest) returns (OrderResponse);
  rpc GetOrder (OrderId) returns (OrderResponse);
  rpc UpdateOrder (OrderUpdateRequest) returns (OrderResponse);
  rpc DeleteOrder (OrderId) returns (DeleteResponse);
}

// Request and Response Messages
message OrderRequest {
  string item = 1;
  int32 quantity = 2;
  double price = 3;
}

message OrderUpdateRequest {
  string id = 1;
  string item = 2;
  int32 quantity = 3;
  double price = 4;
}

message OrderId {
  string id = 1;
}

message OrderResponse {
  string id = 1;
  string item = 2;
  int32 quantity = 3;
  double price = 4;
}

message DeleteResponse {
  string message = 1;
}
